File: lab_8_SQL_FULL_ACCESS/scripts/load-schema.js
/**
 * Load schema.sql into the database (lab_8 Postgres on port 5434).
 */
import "dotenv/config";
import pg from "pg";
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const schemaPath = path.join(__dirname, "..", "schema.sql");

const pool = new pg.Pool({
  host: process.env.PG_HOST || "127.0.0.1",
  port: Number(process.env.PG_PORT || "5434"),
  user: process.env.PG_USER || "sso_user",
  password: process.env.PG_PASSWORD || "sso_pass",
  database: process.env.PG_DATABASE || "sso_db",
});

async function main() {
  if (!fs.existsSync(schemaPath)) {
    throw new Error("schema.sql not found at " + schemaPath);
  }
  const sql = fs.readFileSync(schemaPath, "utf-8");
  console.log("Loading schema from schema.sql...");
  await pool.query(sql);
  console.log("Schema loaded successfully.");
  await pool.end();
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: lab_8_SQL_FULL_ACCESS_PY/scripts/load_schema.py
"""
Load schema.sql into the database (lab_8_PY Postgres on port 5434).
"""
import os
import sys
from pathlib import Path

# Allow running from lab root or from scripts/
lab_root = Path(__file__).resolve().parent.parent
sys.path.insert(0, str(lab_root))

import psycopg2
from dotenv import load_dotenv

load_dotenv(lab_root / ".env")

schema_path = lab_root / "schema.sql"
if not schema_path.exists():
    raise FileNotFoundError(f"schema.sql not found at {schema_path}")

conn = psycopg2.connect(
    host=os.environ.get("PG_HOST", "127.0.0.1"),
    port=int(os.environ.get("PG_PORT", "5434")),
    user=os.environ.get("PG_USER", "sso_user"),
    password=os.environ.get("PG_PASSWORD", "sso_pass"),
    database=os.environ.get("PG_DATABASE", "sso_db"),
)

try:
    print("Loading schema from schema.sql...")
    sql = schema_path.read_text()
    with conn.cursor() as cur:
        for stmt in sql.split(";"):
            stmt = stmt.strip()
            if stmt and not stmt.startswith("--"):
                cur.execute(stmt)
    conn.commit()
    print("Schema loaded successfully.")
finally:
    conn.close()
